### PR TITLE
RDM-7436 initial changes

### PR DESCRIPTION
**Pull request by skandala-cgi**
### JIRA link ###

https://tools.hmcts.net/jira/browse/RDM-7436

### Change description ###
Implement Functional Tests for 'CCD Case Print Service :: GET /jurisdictions/:jid/case-types/:ctid/cases/:cid' operation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: aat/src/resources/features/F-023/F-023.feature [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  And the request [contains a callback which encounters a timeout error],
+  And it is submitted to call the [Print Documents] operation of [CCD Case Print Service API],
+  Then a negative response is received,
+  And the response has all the details as expected.
+
+  @S-004
+  Scenario: Must return a negative response on malformed Case ID
+    Given a user with [an active profile in CCD - wants to see a list of valid documents they can print],
+    When a request is prepared with appropriate values,
+    And the request [contains a malformed caseID],
```
Comment (MSancaktutar) [inline aat/src/resources/features/F-023/F-023.feature:42]: `contains a malformed case ID,`

File: aat/src/java/uk/gov/hmcts/ccd/printservice/befta/CasePrintServiceTestAutomationAdapter.java [hunk 2]
```diff
@@ -12,4 +16,21 @@ public void doLoadTestData() {
...
 
+    @Override
+    public Object calculateCustomValue(BackEndFunctionalTestScenarioContext scenarioContext, Object key) {
+        if (key.toString().startsWith("approximately ")) {
+            try {
+                String actualSize = (String)ReflectionUtils.deepGetFieldInObject(scenarioContext, "testData.actualResponse.body.__fileInBody__.size");
+                String expectedSize = key.toString().replace("approximately ", "");
+                int actualValue = Integer.parseInt(actualSize);
+                int expectedValue = Integer.parseInt(expectedSize);
+                if(Math.abs(actualValue-expectedValue)<50)
```
Comment (tomxelliott) [inline aat/src/java/uk/gov/hmcts/ccd/printservice/befta/CasePrintServiceTestAutomationAdapter.java:27]: Does the coding standard for the repo call for spaces around operators such as `-` and `<`?

File: aat/src/resources/features/F-023/S-005.td.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  "request": {
+    "pathVariables": {
+      "cid": "1111222233334444"
+    }
+    },
+
+  "expectedResponse": {
+    "responseCode": 404,
+    "responseMessage": "Not Found",
```
Comment (MSancaktutar) [inline aat/src/resources/features/F-023/S-005.td.json:18]: Please extent the common 404 response object. That will bring the 2 lines here and the headers below.

File: aat/src/resources/features/F-023/S-006.td.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  "request": {
+    "pathVariables": {
+      "ctid": "$%£@^&*()_+",
+      "cid": "1111222233334444"
+    }
+    },
+
+  "expectedResponse": {
+    "responseCode": 400,
+    "responseMessage": "Bad Request",
```
Comment (MSancaktutar) [inline aat/src/resources/features/F-023/S-006.td.json:19]: Please extent the common 400 response object. That will bring the 2 line shere and the headers below.

File: aat/src/resources/features/F-023/S-007.td.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    "pathVariables": {
+      "ctid": "AAT_AUTH_00",
+      "cid": "1111222233334444"
+    }
+    },
+
+  "expectedResponse": {
+    "responseCode": 404,
+    "responseMessage": "Not Found",
+
```
Comment (MSancaktutar) [inline aat/src/resources/features/F-023/S-007.td.json:20]: Please extent the common 400 response object. That will bring the 2 lines here and the headers below.

File: aat/src/resources/features/F-023/S-008.td.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    "pathVariables": {
+      "jid": "$%£@^&*()_+",
+      "ctid": "FT_MasterCaseType",
+      "cid": "1111222233334444"
+    }
+    },
+
+  "expectedResponse": {
+    "responseCode": 400,
+    "responseMessage": "Bad Request",
```
Comment (MSancaktutar) [inline aat/src/resources/features/F-023/S-008.td.json:20]: Please extent the common 400 response object. That will bring the 2 lines here and the headers below. (no need to then include the headers section at all.)

File: aat/src/resources/features/F-023/S-009.td.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    "pathVariables": {
+      "jid": "ABCDEFGH",
+      "ctid": "FT_MasterCaseType",
+      "cid": "1111222233334444"
+     }
+    },
+
+  "expectedResponse": {
+    "responseCode": 403,
+    "responseMessage": "Forbidden",
```
Comment (MSancaktutar) [inline aat/src/resources/features/F-023/S-009.td.json:20]: Please extent the common 403 response object. That will bring the 2 lines here and the headers below.

Comment (MSancaktutar) [reply aat/src/resources/features/F-023/S-009.td.json:20]: no need for the headers section then.

Comment (skandala-cgi) [reply aat/src/resources/features/F-023/S-009.td.json]: done

File: aat/src/resources/features/F-023/S-008.td.json (hunk 6 above)
Comment (skandala-cgi) [reply aat/src/resources/features/F-023/S-008.td.json]: done

File: aat/src/resources/features/F-023/S-007.td.json (hunk 5 above)
Comment (skandala-cgi) [reply aat/src/resources/features/F-023/S-007.td.json]: done

File: aat/src/resources/features/F-023/S-006.td.json (hunk 4 above)
Comment (skandala-cgi) [reply aat/src/resources/features/F-023/S-006.td.json]: done

File: aat/src/resources/features/F-023/S-005.td.json (hunk 3 above)
Comment (skandala-cgi) [reply aat/src/resources/features/F-023/S-005.td.json]: done

File: aat/src/java/uk/gov/hmcts/ccd/printservice/befta/CasePrintServiceTestAutomationAdapter.java (hunk 2 above)
Comment (skandala-cgi) [reply aat/src/java/uk/gov/hmcts/ccd/printservice/befta/CasePrintServiceTestAutomationAdapter.java]: done

File: aat/src/resources/features/F-023/F-023.feature (hunk 1 above)
Comment (skandala-cgi) [reply aat/src/resources/features/F-023/F-023.feature]: done

File: aat/src/resources/features/F-023/F-023.feature [hunk 8]
```diff
@@ -0,0 +1,92 @@
+@F-023
+Feature: F-023: Print List Of Documents
```
Comment (MSancaktutar) [inline aat/src/resources/features/F-023/F-023.feature:2]: I really couldn't make sense of the operation name `Print List Of Documents`.
Eventually, this api operation is returning a printable document, not itself printing the list of documents.
Could we find out a better name use all around?

File: aat/src/resources/features/F-023/F-023.feature [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    Given a user with [an active profile in CCD - wants to see a list of valid documents they can print],
+    When a request is prepared with appropriate values,
+    And the request [contains a malformed case ID],
+    And it is submitted to call the [Print Documents] operation of [CCD Case Print Service API],
+    Then a negative response is received,
+    And the response has all the details as expected.
+
+  @S-005
+  Scenario: Must return a negative response on non existent Case ID
+    Given a user with [an active profile in CCD - wants to see a list of valid documents they can print],
```
Comment (MSancaktutar) [inline aat/src/resources/features/F-023/F-023.feature:49]: The below line and similar line should change as per the change of the operation, to reflect the aim of the call more directly.
Given a user with [an active profile in CCD - wants to see a list of valid documents they can print],

like...
.. `Given a user with [an active profile in CCD who wants to see the content of a case as a printable document],`
`

File: aat/src/resources/features/F-023/F-023.feature (hunk 8 above)
Comment (skandala-cgi) [reply aat/src/resources/features/F-023/F-023.feature]: updated to "Get Case as Printable Document".

**Pull request merged**